ksql-engine/src/main/java/io/confluent/ksql/codegen/CodeGenRunner.java

```diff
@@ -201,26 +210,55 @@ protected Object visitSubscriptExpression(SubscriptExpression node, Object conte
         throw new RuntimeException(
             "Cannot find the select field in the available fields: " + arrayBaseName);
       }
-      parameterMap.put(
-          schemaField.get().name().replace(".", "_"),
-          SchemaUtil.getJavaType(schemaField.get().schema())
-      );
+      updateFunctionArgTypesAndParams(schemaField.get());
       process(node.getIndex(), context);
       return null;
     }
 
+    private void updateFunctionArgTypesAndParams(final Field schemaField) {
+      final Schema schema = schemaField.schema();
+      functionArguments.addArgumentType(schema.type());
+      parameterMap.put(
+          schemaField.name().replace(".", "_"),
+          new KsqlFunction(schema,
```

Comment:
nit: why do we need to create a dummy function for field references? Can't we just get the type from the schema

Reply:
temp situation so that everything is a function in the map. Makes it easier to deal with later. However, this will change to some other class in next PR as `KsqlFunction` isn't really correct